README.md

```diff
@@ -1,2 +1,462 @@
+
+Recommended minimal-values for a certificate key size, from different
+organizations, vary across time. In the attempt to provide safe settings,
+the default minimal-value for `key_size` will be increased over time.
+
+If you want your certificates to always keep the same `key_size` when
+renewed, set this variable to the desired value.
+
+
+### key_usage
+
+Valid values for `key_usage` are:
+
+* digitalSignature
+* nonRepudiation
+* keyEncipherment
+* dataEncipherment
+* keyAgreement
+* keyCertSign
+* cRLSign
+* encipherOnly
+* decipherOnly
+
+The defaults for `key_usage` are:
+
+* digitalSignature
+* keyEncipherment
+
+### extended_key_usage
+
+Any valid oid can be used to set one or more `extended_key_usage`.
+In addition to that there is also a list of known aliases that
+will be recognized by the role:
+
+* id-kp-serverAuth
+* id-kp-clientAuth
+* id-kp-codeSigning
+* id-kp-emailProtection
+* id-kp-timeStamping
+* id-kp-OCSPSigning
+* id-kp-ipsecEndSystem
+* id-kp-ipsecTunnel
+* id-kp-ipsecUser
+
+If `extended_key_usage` is not set the role will default to:
+
+* id-kp-serverAuth
+* id-kp-clientAuth
+
+### run hooks
+
+Sometimes you need to execute a command just before a certificate is
+renewed and another command just after. In order to do that use
+`run_before` and `run_after`.
+
+The value provided to `run_before` and `run_after` will be wrapped and
+stored in shell script files that later will be executed by the provider.
+
+
+## CAs and Providers
+
+| CA               | Providers   | CA description                                  | Requirements                                    |
+|------------------|-------------|-------------------------------------------------|-------------------------------------------------|
+| self&#x2011;sign | certmonger* | Issue self-signed certificates from a local CA. |                                                 |
+| ipa              | certmonger* | Issue certificates using the FreeIPA CA.        | Host needs to be enrolled in a FreeIPA server.  |
+
+ *\* Default provider.*
+
+CA represents the CA certificates that will be used to issue and sign the
+requested certificate. Provider it's the method used to send the certificate
+request to the CA and then retrieve the signed certificate.
+
+If a user chooses `self-sign` CA, with `certmonger` as provider and, later on
+decide to change the provider to `openssl`, the CA certificates used in both
+cases needs to be the same. *Please note that `openssl` is **not yet a supported**
+provider and it's only mentioned here as an example.*
+
+
+## Examples:
+
+
+### Issue a self-signed certificate:
+
+Issue a certificate for `*.example.com` and place it in the standard
+directory for the distribution.
+
+
+```yaml
+
+---
+- hosts: webserver
+
+  vars:
+    certificate_requests:
+      - name: mycert
+        dns: *.example.com
+        ca: self-sign
+
+  roles:
+    - certificate
+```
+
+The directories for each distribution are:
+
+* Debian/Ubuntu:
```

Comment:
Is this role supported on Debian/Ubuntu?  If so, do we need to add that to platform support?

Reply:
Added a note for Debian 10.